src/votekit/outcome_model.py

```diff
@@ -0,0 +1,134 @@
+from pydantic import BaseModel
+from .profile import PreferenceProfile
+from typing import Optional
+
+
+# 1. is add_winners_and_losers necessary? --A: probabably not
+# 2. do we want anything more?]
+# 3. RCV cruncher: inspiration for functions--> statistics functions/things
+# Example of immutable data model for results
+
+
+class Outcome(BaseModel):
+    """
+    curr_round (an Int): current round number
+    elected (a list of Candidate): candidates who pass a certain threshold to win an election
+    eliminated (a list of Candidate): candidates who were eliminated (lost in the election)
+    remaining (a list of Candidate): candidates who are still in the running
+    rankings (a list of a set of Candidate): ranking of candidates with sets representing ties
+    profile (a PreferenceProfile): a list of ballot types
+    (schedule preferences) and their frequency of times
+    winners_votes (a Dict of candidates and list of their ballots):
+    each winner's list of ballots that elected them
+    previous: an instance of Outcome representing the previous round
+    """
+
+    curr_round: int
+    elected: list[str] = []
+    eliminated: list[str] = []
+    remaining: list[str] = []
+    profile: Optional[PreferenceProfile] = None
```

Comment:
Since we're already returning profiles when we run elections, I don't think we need to store them in outcome class as well

Reply:
I think I disagree. Most often our workflow is gonna be geared towards accessing the outcome via the election, but sometimes, long after we've run elections and have a bunch of outcomes we might wanna do some analysis with the outcomes and be able to access the corresponding profiles. We're not storing the outcome as an attribute of the profile; we should at least have the option of storing the profile as an attribute of the outcome.

Reply:
I can however change the code so that we store profile only if it has not been stored in any previous round. That should take care of overuse of memory etc